test/e2e/e2e_suite_test.go

```diff
@@ -31,6 +31,13 @@ var _ = BeforeSuite(func() {
 			By("Check Cluster health")
 			err := e2e.CheckClusterHealth(clusterContext)
 			if err != nil {
+				// collect diagnostic details for workload cluster
+				log.Println("Collecting workload cluster diagnostic details in /tmp folder...")
+				_, err =utils.GetWorkloadClusterDiagnostics(e2e.ConfigVal.GuestClusterName)
```

Comment:
There's a lint error in the CI due to this

```suggestion
				_, err = utils.GetWorkloadClusterDiagnostics(e2e.ConfigVal.GuestClusterName)
```